### PR TITLE
Document icu_calendar

### PR DESCRIPTION
**Pull request by Manishearth**
One thing I didn't do is rename "ethiopic" to "ethiopian" in our types and modules. There's no such thing as an "Ethiopic calendar", it's called the Ethiopian calendar, and the BCP47 string is "ethiopic" to stay within length restrictions.

### CODE REVIEW

File: components/calendar/src/ethiopic.rs [hunk 1]
```diff
@@ -56,11 +54,28 @@ pub enum EthiopicEraStyle {
...
+///
+/// It can be constructed in two modes: using the Amete Alem era scheme, or the Amete Mihret era scheme (the default)
+///
+/// [Ethiopian calendar]: https://en.wikipedia.org/wiki/Ethiopian_calendar
+///
+/// # Era codes
+///
+/// This calendar supports three era codes, based on what mode it is in. In the Amete Mihret scheme it has
+/// the `"incar"` and `"pre-incar"` eras, 1 Incarnation is 9 CE. In the Amete Alem scheme, it instead has a single era,
+/// `"mundi`, where 1 Anno Mundi is 5493 BCE. Dates before that use negative year numbers.
```
Comment (robertbastian) [inline components/calendar/src/ethiopic.rs:73]: Link to `EthiopicEraStyle` for more information.

File: components/calendar/src/ethiopic.rs [hunk 2]
```diff
@@ -56,11 +54,28 @@ pub enum EthiopicEraStyle {
     /// in this calendar
     AmeteAlem,
 }
-/// The Ethiopic Calendar
```
Comment (robertbastian) [inline components/calendar/src/ethiopic.rs:59]: The docs now use both "ethiopic" and "ethiopian"

File: components/calendar/src/japanese.rs [hunk 3]
```diff
@@ -45,25 +45,64 @@
...
+///
+/// This type can be used with [`Date`] or [`DateTime`] to represent dates in this calendar.
+///
+/// [Japanese calendar]: https://en.wikipedia.org/wiki/Japanese_calendar
+///
+/// # Era codes
+///
+/// This calendar currently supports seven era codes. It supports the five post-Meiji eras
+/// (`"meiji"`, `"taisho"`, `"showa"`, `"heisei"`, `"reiwa"`), as well as using the Gregorian
+/// `"bce"` and `"ce"` for dates before the Meiji era.
```
Comment (robertbastian) [inline components/calendar/src/japanese.rs:69]: ... For all eras, see [`JapaneseExtended`].

File: components/calendar/src/lib.rs [hunk 4]
```diff
@@ -167,10 +170,10 @@ pub mod japanese;
 pub mod julian;
 pub mod provider;
 pub mod types;
+pub mod week_of;
 
 pub use any_calendar::{AnyCalendar, AnyCalendarKind};
 pub use calendar::Calendar;
-pub use calendar_arithmetic::{ArithmeticDate, CalendarArithmetic};
 pub use date::{AsCalendar, Date, Ref};
```
Comment (robertbastian) [inline components/calendar/src/lib.rs:177]: `Ref` has its whole doc in one paragraph which takes up a lot of space in the docs:
image

**Review by robertbastian — COMMENTED** [review]
Just some nits

File: components/calendar/src/ethiopic.rs (hunk 2 above)
Comment (Manishearth) [reply components/calendar/src/ethiopic.rs:59]: I plan to do a full upgrade later, we should _always_ be saying ethiopian except in the BCP47 tag

**Review by robertbastian — APPROVED** [review]

**Pull request merged**